features/provider/password/reset.feature

```diff
@@ -0,0 +1,78 @@
+@javascript
+Feature: Provider password reset
+  In order to sign in even if I forgot my password
+  As a user
+  I should be able to reset it
+
+  Background:
+    Given a provider
+    And an active user "Pepe" of account "foo.3scale.localhost" with email "pepe@example.com"
+    And they go to the provider reset password page
```

Comment:
For the first two scenarios ("Reset password on provider domain" and "Reset password on master domain") the following background steps are not needed:
- And they go to the provider reset password page
- And an active user "Pepe" of account "foo.3scale.localhost" with email "pepe@example.com"

In fact, it is quite confusing, especially for the master domain scenario.

Maybe we can split these into different Rules, and only apply these background steps to the scenarios that actually depend on them.